src/opera_disp_tms/generate_sw_disp_tile.py

```diff
@@ -0,0 +1,233 @@
+            utils.wkt_from_epsg(4326),
+            in_granule['spatial_ref'].attrs['crs_wkt'],
+        )
+        ref_value = in_granule.sel(x=ref_x, y=ref_y, method='nearest').data.item()
+        in_granule -= ref_value
+        in_granule.attrs['reference_point_array'] = frame.reference_point_array
+        in_granule.attrs['reference_point_geo'] = frame.reference_point_geo
+
+    return in_granule
+
+
+def create_blank_copy_tile(input_path: Path, output_path: Path) -> None:
+    """Create a blank copy of a GeoTiff file
+
+    Args:
+        input_path: Path to the input GeoTiff file
+        output_path: Path to the output GeoTiff file
+    """
+    ds = gdal.Open(str(input_path))
+    metadata = ds.GetMetadata()
+    transform = ds.GetGeoTransform()
+    projection = ds.GetProjection()
+    x_size = ds.RasterXSize
+    y_size = ds.RasterYSize
+    band = ds.GetRasterBand(1)
+    data = band.ReadAsArray().astype(float)
+    data[:] = np.nan
+    ds = None
+
+    driver = gdal.GetDriverByName('GTiff')
+    opts = ['TILED=YES', 'COMPRESS=LZW', 'NUM_THREADS=ALL_CPUS']
+    out_ds = driver.Create(str(output_path), x_size, y_size, 1, gdal.GDT_Float32, options=opts)
+
+    out_band = out_ds.GetRasterBand(1)
+    out_band.SetNoDataValue(np.nan)
+    out_band.WriteArray(data)
+    out_ds.SetGeoTransform(transform)
+    out_ds.SetProjection(projection)
+    out_ds.SetMetadata(metadata)
+
+    out_band.FlushCache()
+    out_ds = None
+
+
+def create_product_name(metadata_name: str, begin_date: datetime, end_date: datetime) -> str:
+    """Create a product name for a short wavelength cumulative displacement tile
+    Takes the form: SW_CUMUL_DISP_YYYYMMDD_YYYYMMDD_ORBIT_BBOX.tif
+
+    Args:
+        metadata_name: The name of the metadata file
+        begin_date: The beginning of the date range to generate the tile for
+        end_date: The end of the date range to generate the tile for
+    """
+    parts = metadata_name.split('_')[1:]
+    date_fmt = '%Y%m%d'
+    begin_date = datetime.strftime(begin_date, date_fmt)
+    end_date = datetime.strftime(end_date, date_fmt)
+    name = '_'.join(['SW_CUMUL_DISP', begin_date, end_date, *parts])
+    return name
+
+
+def create_sw_disp_tile(metadata_path: Path, begin_date: datetime, end_date: datetime) -> Path:
+    """Create a short wavelength cumulative displacement tile
+
+    Args:
+        metadata_path: Path to the metadata GeoTiff file
+        begin_date: The beginning of the date range to generate the tile for
+        end_date: The end of the date range to generate the tile for
+
+    Returns:
+        Path to the generated tile
+    """
+    if not metadata_path.exists():
+        raise FileNotFoundError(f'{metadata_path} does not exist')
+    if begin_date > end_date:
+        raise ValueError('Begin date must be before end date')
+
+    product_name = create_product_name(metadata_path.name, begin_date, end_date)
+    product_path = metadata_path.parent / product_name
```

Comment:
This surprises me. I'd expect the output to be created in my current directory, not adjacent to the metadata geotiff I specified.